apps/provisioning/pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -256,6 +256,10 @@ type RepositorySpec struct {
 	// The repository on GitLab.
 	// Mutually exclusive with local | github | git.
 	GitLab *GitLabRepositoryConfig `json:"gitlab,omitempty"`
+
+	// The connection the repository references.
+	// This means the Repository is interacting with git via a Connection.
+	Connection *string `json:"connection,omitempty"`
```

Comment:
It's not clear it's the name of the connection. Maybe can we can do: 

```
connection
  name: connection-id
```

in this format, we will be able to add other things to the connection configuration if needed in the future (e.g. namespace (won't happen), labels (?), etc).

Reply:
Done!